src/os/linux/xdg.rs

```diff
@@ -0,0 +1,412 @@
+                reading_quoted = false;
+            }
+        // If we're reading a singly quoted argument ('like this')
+        } else if reading_singlequoted {
+            // Same thing but for '
+            if c != b'\'' {
+                text_atom.push(c);
+            } else {
+                // When we find another ', we collected a text atom
+                // If there is text we store it
+                if text_atom.len() > 0 {
+                    let text_atom_string = parse_quoted_text(
+                        OsStr::from_bytes(&text_atom),
+                        argument,
+                        icon,
+                        desktop_entry_path,
+                    );
+                    parsed_command_parts.push(text_atom_string);
+                    text_atom.clear();
+                }
+                // And the quoting ended
+                reading_singlequoted = false;
+            }
+        // If not quoting, or scaping, then space is a text atom separator
+        } else if [b' ', b'\t', b'\n'].contains(&c) {
+            // If there is text we store it
+            if text_atom.len() > 0 {
+                let text_atom_string = parse_unquoted_text(
+                    OsStr::from_bytes(&text_atom),
+                    argument,
+                    icon,
+                    desktop_entry_path,
+                );
+                parsed_command_parts.push(text_atom_string);
+                text_atom.clear();
+            }
+        // If a non whitespace, nor backslash character, when we're neither escaping nor in quotes, then...
+        } else {
+            match c {
+                b'"' => reading_quoted = true,
+                b'\'' => reading_singlequoted = true,
+                anything_else => text_atom.push(anything_else),
+            }
+        }
+    } // End of iteration over the command's bytes
+
+    // At the end of the loop we flush whatever was being accumulated to the command parts
+    if text_atom.len() > 0 {
+        // If the value was well formed, quoted strings end on a quote character, and
+        // not on EOF, so this should be unquoted.
+        let text_atom_string = parse_unquoted_text(
+            OsStr::from_bytes(&text_atom),
+            argument,
+            icon,
+            desktop_entry_path,
+        );
+        parsed_command_parts.push(text_atom_string);
+        text_atom.clear();
+    }
+
+    log::debug!(
+        "XDG parsed command {:?} to {:?}",
+        command,
+        parsed_command_parts
+    );
+    parsed_command_parts
+}
+
+// Returns a vector of all the relevant xdg desktop application entries
+// Check out:
+// https://wiki.archlinux.org/index.php/XDG_Base_Directory
+// That explains the default values and the relevant variables.
+pub fn get_xdg_data_dirs() -> Vec<PathBuf> {
+    let mut result = Vec::new();
+
+    if let Ok(home) = crate::util::home_dir() {
+        let home: PathBuf = home.into();
```

Comment:
```suggestion
```
`home` is already a `PathBuf` now, thanks to the slight convenience of `util::home_dir`!